src/components/FilterOverlay.vue

```diff
@@ -159,8 +162,17 @@ export default defineComponent({
           const aqlNetwork: Network = promise[0];
 
           if (aqlNetwork.nodes.length !== 0) {
-          // Update state with new network
-            store.dispatch.updateNetwork({ network: aqlNetwork });
+            const nodes = defineNeighbors(aqlNetwork.nodes, aqlNetwork.edges as Edge[]);
+
+            // Build the network object and set it as the network in the store
+            const aqlNetworkElements: Network = {
+              nodes: nodes as Node[],
+              edges: aqlNetwork.edges as Edge[],
+            };
+
+            // Update state with new network
+            store.commit.setNetworkOnLoad(aqlNetworkElements);
+            store.dispatch.updateNetwork({ network: aqlNetworkElements });
```

Comment:
If you accept the other change in this file, these will need to be switched and you'll need to pass another network to `setNetworkOnLoad`
```suggestion
            store.dispatch.updateNetwork({ network: aqlNetworkElements });
            store.commit.setNetworkOnLoad(store.state.network);
```